src/libDirectoryService/FinalBlockPreProcessing.cpp

```diff
@@ -735,8 +739,12 @@ bool DirectoryService::CheckMicroBlockInfo() {
     //                 << "; hashes: " << hashesInMicroBlocks[i]
     //                 << "; IsMicroBlockEmpty: "
     //                 << m_finalBlock->GetIsMicroBlockEmpty()[i]);
-    auto& microBlocks = m_microBlocks[m_mediator.m_currentEpochNum];
-    for (auto& microBlock : microBlocks) {
+    std::set<MicroBlock> microBlocks;
+    if (m_microBlocks.find(m_mediator.m_currentEpochNum) !=
```

Comment:
Same as previous comment

Reply:
Now changed to if not found micro block info, then continue to check next next one.